### PR TITLE
Refactor: Invert dependency between query history and remote quries managers

### PR DESCRIPTION
**Pull request by charisk**
Currently, the `RemoteQueriesManager` depends on the `QueryHistoryManager` and related models e.g. `QueryHistoryInfo`. This is a bit problematic because the `RemoteQueriesManager` ends up having knowledge about query history stuff and can receive events that are potentially for local queries. 

This PR inverts that dependency and makes `RemoteQueriesManager` a dependency of QueryHistoryManager. The `RemoteQueriesManager` is completely independent of query history and simply publishes events that the `QueryHistoryManager` can consume. I think this way makes more sense because the query history area **has** to be aware of local/remote queries, but remote queries don't have to necessarily be aware of the query history.

## Checklist
N/A
- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/query-history.ts [hunk 1]
```diff
@@ -307,21 +311,11 @@ export class QueryHistoryManager extends DisposableObject {
...
-    ._onDidRemoveQueryItem.event;
-
-  private readonly _onWillOpenQueryItem = super.push(new EventEmitter<QueryHistoryInfo>());
-  readonly onWillOpenQueryItem: Event<QueryHistoryInfo> = this
-    ._onWillOpenQueryItem.event;
-
   constructor(
     private readonly qs: QueryServerClient,
     private readonly dbm: DatabaseManager,
+    private readonly localQueriesInterfaceManager: InterfaceManager,
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/query-history.ts:317]: We _could_ rename `InterfaceManager` to something like `LocalQueriesInterfaceManager`.

File: extensions/ql-vscode/src/query-history.ts [hunk 2]
```diff
@@ -548,12 +543,47 @@ export class QueryHistoryManager extends DisposableObject {
...
+    this.remoteQueriesManager.onRemoteQueryAdded(event => {
+      const historyItem: RemoteQueryHistoryItem = {
+        t: 'remote',
+        status: QueryStatus.InProgress,
+        completed: false,
+        queryId: event.queryId,
+        remoteQuery: event.query,
+      };
+
+      this.addQuery(historyItem);
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/query-history.ts:556]: Minor:

```suggestion
      this.addQuery({
        t: 'remote',
        status: QueryStatus.InProgress,
        completed: false,
        queryId: event.queryId,
        remoteQuery: event.query,
      });
```

File: extensions/ql-vscode/src/query-history.ts [hunk 3]
```diff
@@ -548,12 +543,47 @@ export class QueryHistoryManager extends DisposableObject {
     );
   }
 
+  private registerToRemoteQueriesEvents() {
+    this.remoteQueriesManager.onRemoteQueryAdded(event => {
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/query-history.ts:547]: Pretty sure these need to be added to the subscription for disposal. All of the `onRemoteQuery*` event handlers.

**Review by aeisenberg — COMMENTED** [review]
Looks sensible.

File: extensions/ql-vscode/src/query-history.ts (hunk 3 above)
Comment (charisk) [reply extensions/ql-vscode/src/query-history.ts]: Good catch! Sorted.

**Comment by charisk** [comment]
> Thanks for tidying this up! 🧹
> 
> I tried locally, and I wonder if something has changed with reporting progress of the `Run Variant Analysis` command? 🔄
> 
> Specifically, the final stage of the command ("Sending request") now stays on-screen the whole time until the results are back. I'd expect it to disappear as soon as we've successfully scheduled the run:
> 
> image

Great catch! Not sure how I missed this.

I'm not sure what could be causing this. I'll take a look tomorrow. @aeisenberg please let me know if you have any ideas. I don't think I've changed the "run remote query" code path so I'm a bit puzzled as to why this would be happening.

**Comment by charisk** [comment]
Bug fixed now - there was an `await` on the execution of the monitor command and things just _happened_ to work before my changes.

**Review by elenatanasoiu — APPROVED** [review]
This makes sense 👍

**Pull request merged**